Eventy/src/main/java/org/example/eventy/solutions/repositories/SolutionRepository.java

```diff
@@ -26,7 +30,50 @@ Page<Solution> findUsersFavoriteSolutions(@Param("userId") Long userId,
                                         @Param("search") String search,
                                         Pageable pageable);
 
-    @Query("SELECT COUNT(s) FROM Solution s WHERE s.id IN " +
-            "(SELECT us.id FROM User u JOIN u.favoriteSolutions us WHERE u.id = :userId)")
+    @Query("""
+           SELECT COUNT(s) FROM Solution s WHERE s.id IN (
+                SELECT us.id FROM User u JOIN u.favoriteSolutions us WHERE u.id = :userId)
+           """)
     long countUsersFavoriteSolutions(@Param("userId") Long userId);
+
+    // FALI TYPE OVDEEE
+    @Query("""
+           SELECT s
+           FROM Solution s
+           LEFT JOIN s.eventTypes et
+           WHERE (:search = '' OR s.name ILIKE ('%' || :search || '%'))
+             AND (:categories IS NULL OR s.category.name IN :categories)
+             AND (:eventTypes IS NULL OR EXISTS (
+               SELECT 1
+               FROM s.eventTypes et
+               WHERE et.name IN :eventTypes
+               AND SIZE(s.eventTypes) = :eventTypesSize
+             ))
+             AND (:company = '' OR s.provider.name = :company)
+             AND (:minPrice IS NULL OR :maxPrice IS NULL OR (s.price BETWEEN :minPrice AND :maxPrice))
+             AND (s.isAvailable = :isAvailable)
+             AND (s.isVisible = TRUE)
+             AND (s.isDeleted = FALSE)
+             AND NOT EXISTS (
+               SELECT r
+               FROM Reservation r
+               WHERE r.selectedService.id = s.id
+                 AND (:startDate <= r.reservationEndDateTime AND :startDate >= r.reservationStartDateTime)
+                 AND (:endDate >= r.reservationStartDateTime AND :endDate <= r.reservationEndDateTime))
```

Comment:
Maybe make it an OR here? Ill write where:
SELECT r
                FROM Reservation r
                WHERE r.selectedService.id = s.id
                  AND (:startDate <= r.reservationEndDateTime AND :startDate >= r.reservationStartDateTime)
                  ==> HERE => OR (:endDate >= r.reservationStartDateTime AND :endDate <= r.reservationEndDateTime)
                  
Why? It should return a reservation if it overlaps either with startDate or the endDate. Example: as you said let's have it: reservationStartDateTime = 26.12.2024, reservationEndDateTime = 29.12.2024, startDate = 25.12.2024., endDate = 27.12.2024.

:startDate <= r.reservationEndDateTime AND :startDate >= r.reservationStartDateTime will return false here, since :startDate is less than reservationStartDateTime

This means that the whole where clause will be false, since it is connected with AND. So that part between start and end date checking should be an OR? Tell me your thoughts too!